core/src/test/java/kafka/clients/consumer/ConsumerIntegrationTest.java

```diff
@@ -185,4 +194,72 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
             }
         }
     }
+
+    @ClusterTest(types = {Type.KRAFT}, brokers = 3)
+    public void testLeaderEpoch(ClusterInstance clusterInstance) throws Exception {
+        String topic = "test-topic";
+        clusterInstance.createTopic(topic, 1, (short) 2);
+        var msgNum = 10;
+        sendMsg(clusterInstance, topic, msgNum);
+
+        try (var consumer = clusterInstance.consumer()) {
+            List<TopicPartition> topicPartitions = Collections.singletonList(new TopicPartition(topic, 0));
+            consumer.assign(topicPartitions);
+            consumer.seekToBeginning(Collections.singletonList(new TopicPartition(topic, 0)));
+
+            int consumed = 0;
+            while (consumed < msgNum) {
+                ConsumerRecords<Object, Object> records = consumer.poll(Duration.ofMillis(1000));
+                for (ConsumerRecord<Object, Object> record : records) {
+                    assertTrue(record.leaderEpoch().isPresent());
+                    assertEquals(0, record.leaderEpoch().get());
+                }
+                consumed += records.count();
+            }
+
+            // make the leader epoch increment by shutdown the leader broker
+            shutdownFirstPartitionLeader(clusterInstance, topic);
+
+            sendMsg(clusterInstance, topic, msgNum);
+
+            consumed = 0;
+            while (consumed < msgNum) {
+                ConsumerRecords<Object, Object> records = consumer.poll(Duration.ofMillis(1000));
+                for (ConsumerRecord<Object, Object> record : records) {
+                    assertTrue(record.leaderEpoch().isPresent());
+                    assertEquals(1, record.leaderEpoch().get());
+                }
+                consumed += records.count();
+            }
+        }
+    }
+
+    private void shutdownFirstPartitionLeader(ClusterInstance clusterInstance,
+                                              String topic) throws Exception {
+        var leaderBrokerId = -1;
+        try (var admin = clusterInstance.admin()) {
+            DescribeTopicsResult result = admin.describeTopics(List.of(topic));
```

Comment:
finding the leader is a common op, so could you please add a helper to `ClusterInstance` to return leader of partition?